src/plugin/file/zip/ui/zipimport.js

```diff
@@ -0,0 +1,310 @@
+
+
+/**
+ * The ZIP import directive; use the Wizard Directive
+ *
+ * @return {angular.Directive}
+ */
+plugin.file.zip.ui.zipImportDirective = function() {
+  return {
+    restrict: 'E',
+    replace: true,
+    templateUrl: os.ROOT + 'views/plugin/zip/zipimport.html',
+    controller: plugin.file.zip.ui.ZIPImportCtrl,
+    controllerAs: 'ctrl'
+  };
+};
+os.ui.Module.directive('zipimport', [plugin.file.zip.ui.zipImportDirective]);
+
+
+/**
+ * Controller for the ZIP import dialog
+ *
+ * @param {!angular.Scope} $scope
+ * @param {!angular.JQLite} $element
+ * @param {!angular.$timeout} $timeout
+ * @param {!Object.<string, string>} $attrs
+ * @extends {os.ui.file.FileImportCtrl}
+ * @constructor
+ * @ngInject
+ */
+plugin.file.zip.ui.ZIPImportCtrl = function($scope, $element, $timeout, $attrs) {
+  plugin.file.zip.ui.ZIPImportCtrl.base(this, 'constructor', $scope, $element, $timeout);
+
+  /**
+   * @type {angular.$timeout|null}
+   * @private
+   */
+  this.timeout__ = $timeout;
+
+  /**
+   * @type {os.ui.im.ImportManager}
+   * @private
+   */
+  this.im_ = os.ui.im.ImportManager.getInstance();
+
+  /**
+   * @type {!Array<osx.import.FileWrapper>|null}
+   * @private
+   */
+  this.importers_;
+
+  /**
+   * Group of objects related to each step in the chain of importers
+   *
+   * @type {osx.import.FileWrapper|null}
+   * @private
+   */
+  this.curImporter_ = null;
+
+  /**
+   * Group of objects related to each step in the chain of importers
+   *
+   * @type {function()|null}
+   * @private
+   */
+  this.curListener_ = null;
+
+  /**
+   * Group of objects related to each step in the chain of importers
+   *
+   * @type {angular.$q.Promise|null}
+   * @private
+   */
+  this.curTimeout_ = null;
+
+  /**
+   * @type {plugin.file.zip.ZIPParserConfig}
+   * @private
+   */
+  this.config_ = /** @type {plugin.file.zip.ZIPParserConfig} */ ($scope['config']);
+
+  /**
+   * @type {number}
+   * @private
+   */
+  this.wait_ = 0;
+
+  /**
+   * @type {Array.<osx.import.FileWrapper>|null}
+   */
+  this['files'] = this.config_['files'];
+
+  /**
+   * @type {boolean}
+   */
+  this['loading'] = false;
+
+  /**
+   * @type {boolean}
+   */
+  this['valid'] = false;
+
+  this.validate_();
+};
+goog.inherits(plugin.file.zip.ui.ZIPImportCtrl, os.ui.file.FileImportCtrl);
+
+
+/**
+ * @export
+ */
+plugin.file.zip.ui.ZIPImportCtrl.prototype.finish = function() {
+  var entries = this['files'];
+
+  if (!entries) {
+    var msg = 'No files selected to import from ZIP';
+    os.alert.AlertManager.getInstance().sendAlert(msg, os.alert.AlertEventSeverity.WARNING);
+    return;
+  }
+
+  this.importers_ = [];
+  var unsupported = {};
+
+  for (var i = 0; i < entries.length; i++) {
+    var entry = entries[i];
+    if (entry.enabled === true) {
+      var file = entry.file;
+      var type = file.getType();
+      var ui = this.im_.getImportUI(type);
+      if (ui) {
+        var importer = /** @type osx.import.FileWrapper */ ({
```

Comment:
If you move the `FileWrapper` typedef into `src/`, then it will not be necessary to use quotes here. Newer syntax rules also let you do this: `{file, ui}`, which is equivalent to `{file: file, ui: ui}`, but I'm not sure when that was added so the compiler may yell at you.